cylc/flow/cycling/iso8601.py

```diff
@@ -633,125 +636,165 @@ def _get_old_anchor_step_recurrence(anchor, step, start_point):
     return str(anchor_point) + "/" + str(step)
 
 
-def ingest_time(value, my_now=None):
-    """
-    Allows for relative and truncated cycle points,
-    and cycle point as an offset from 'now'
-    """
+def ingest_time(value, now=None):
+    """Handle relative, truncated and prev/next cycle points.
 
-    # Send back integer cycling, date-only, and expanded datetimes.
-    if re.match(r"\d+$", value):
-        # Could be an old date-time cycle point format, or integer format.
-        return value
-    if (value.startswith("-") or value.startswith("+")) and "P" not in value:
-        # Expanded year
-        return value
+    Args:
+        value (str):
+            The string containing the prev()/next() stuff.
+        now (metomi.isodatetime.data.TimePoint):
+            A time point to use as the context for resolving the value.
 
+    """
+    # remove extraneous whitespace from cycle point
+    value = value.replace(" ", "")
     parser = SuiteSpecifics.point_parser
-    offset = None
+    timepoint = parser.parse(value)
+
+    # integer point or old-style date-time cycle point format
+    is_integer = bool(re.match(r"\d+$", value))
+    # iso8601 expanded year
+    is_expanded = (
+        (value.startswith("-") or value.startswith("+"))
+        and "P" not in value
+    )
+    # prev() or next()
+    is_prev_next = "next" in value or "previous" in value
+    # offset from now (±P...)
+    is_offset = value.startswith("P") or value.startswith("-P")
+    # missing date-time components off the front (e.g. 01T00)
+    is_truncated = timepoint.truncated
+
+    if (
+        is_integer
+        or is_expanded
+        or not any((is_prev_next, is_offset, is_truncated))
```

Comment:
This is the line which fixes the bug, allowing the code to return earlier skipping the whole `now` logic (which isn't needed).